en/docs/references/adaptive-authentication-js-api-reference.md

```diff
@@ -0,0 +1,641 @@
+# Adaptive Authentication JS API Reference
+
+With adaptive authentication, it is possible to configure dynamic
+sequences based on runtime parameters such as the user’s IP address,
+user role, etc. WSO2 Identity Server allows you to define a dynamic
+authentication sequence using authentication scripts written in
+JavaScript.
+
+For more information on adaptive authentication, see [Adaptive Authentication](../../../guides/adaptive-auth/configure-adaptive-auth/)
+
+The following sections present the core API reference for the
+JavaScript-based conditional authentication functions and fields.
+
+!!! tip
+    As the authentication script is designed as a loosely typed, functional
+    language similar to JavaScript, common knowledge on JavaScript may help
+    you to compose effective authentication scripts.
+
+---
+
+## Core functions
+
+##### onLoginRequest(context)
```

Comment:
shall we make these h3?

Reply:
I tried it and it appears too big and odd since this page has very less content under each heading.